main.js

```diff
@@ -324,6 +355,103 @@ define(function (require, exports, module) {
         toggleCmd.setChecked(visible);
     }
 
+    // Support custom theme
+    function _onSelectCSS(err, fi) {
+        if (err) {
+            console.error(err);
+            Dialogs.showModalDialog(
+                "",
+                "Markdown Preview",
+                "There was an error with your selection."
+            );
+            return;
+        }
+
+        if (fi.length === 1) {
+            _prefs.set("theme", "custom");
+            _prefs.set("customTheme", fi[0]);
+            _updateSettings();
+        } else {
+            _showSettings();
+        }
+    }
+
+    function _onSelectModule(id) {
+        if (id === "cancel") {
+            _showSettings();
+            return;
+        }
+
+        FileSystem.showOpenDialog(
+            false,
+            false,
+            "Select a CSS file",
+            brackets.app.getUserDocumentsDirectory(),
+            ["css"],
+            _onSelectCSS
+        );
+    }
+
+    function setCustomTheme() {
+        _hideSettings();
+
+        Dialogs.showModalDialog(
```

Comment:
This dialog is not necessary. Selecting the 'Custom' theme should go directly to the file system open dialog.